Docs/CollectionExtensions.md

```diff
@@ -0,0 +1,4 @@
+# CollectionExtensions
+
+Coming Soon!
+
```

Comment:
I guess that it's good to have room to put in documentation later, but are you sure that it's not better to put it in now?

Not really a Pull-Request breaker, but I think it's worth getting everything done at once if possible so that we know what somethings intended function is when it comes to reviewing the code.

Reply:
Didn't think they were part of this PR.

https://github.com/dubit/duck-utils/pull/7/files

This old PR (merged) should contain those changes